### PR TITLE
HYPR-48 RPi4: PCIe on guest VM

### PR DESCRIPTION
**Pull request by jsvapiav**
* Pass USB over PCIe
* Move VM0 to 1st Gb of RAM
* Pass communication with VC and reset USB nodes to VM0
* Pass PCIe MAP memory to VM0
* Add all PCIe IRQs to be able to work both with legacy and MSI IRQs

### CODE REVIEW

**Comment by hlyytine** [comment]
Looks good, check my suggestions, though.

File: apps/Arm/vm_qemu_virtio/rpi4/devices.camkes [hunk 1]
```diff
@@ -51,41 +51,58 @@ assembly {
             "dtb_name" : "",
         };
         vm0.mmios = [
-            "0xff846000:0x1000:12"
+            "0xff846000:0x1000:12",
```
[the remark this answers is not in the document]
Comment (jsvapiav) [reply apps/Arm/vm_qemu_virtio/rpi4/devices.camkes]: Removed.

**Comment by hlyytine** [comment]
Looks good now.

File: apps/Arm/vm_qemu_virtio/rpi4/devices.camkes [hunk 2]
```diff
@@ -51,41 +51,58 @@ assembly {
...
             { "path": "/soc/cprman@7e101000" },
             { "path": "/soc/watchdog@7e100000" },
             { "path": "/v3dbus/v3d@7ec04000" },
+
+            // PCIe
+            { "path": "/scb/pcie@7d500000" },
+            { "path": "/scb/pcie@7d500000/pci@1,0" },
+
+            // USB
+            // Here need to be following line,
```
[the remark this answers is not in the document]
Comment (jsvapiav) [reply apps/Arm/vm_qemu_virtio/rpi4/devices.camkes]: Add `TODO` mark and link to ticket for tracking.

File: apps/Arm/vm_qemu_virtio/rpi4/devices.camkes [hunk 3]
```diff
@@ -51,41 +51,58 @@ assembly {
...
+            "0x600000000:33", // PCIe
         ];
-        vm0.irqs =  [];
         vm0.dtb = dtb([
             { "path": "/soc/mailbox@7e00b880" },
             { "path": "/soc/serial@7e215040" },
             { "path": "/soc/cprman@7e101000" },
             { "path": "/soc/gpio@7e200000" },
             { "path": "/scb/ethernet@7d580000" },
+
```
[the remark this answers is not in the document]
Comment (jsvapiav) [reply apps/Arm/vm_qemu_virtio/rpi4/devices.camkes]: Ok, removed.

**Pull request merged**